openmc/deplete/cram.py

```diff
@@ -68,18 +68,27 @@ def __call__(self, A, n0, dt):
             material or an atom density
         dt : float
             Time [s] of the specific interval to be solved
+        direct_solve : bool
+            Whether to use scipy.linalg.spsolve directly. Default is False
+            which performs an explicit LU factorization.
```

Comment:
If LU factorization is better, why not use that all the time?

Reply:
Sorry , that came out before the rest of the review. So there's some context missing in my comment.

But it also doesn't appear that this is configurable by the user. We pass   `n_result = func(matrix, n_multi, dt, direct_solve=True)`  later on. So how would a user be able to configure this setting?

From the [Zen of Python / PEP 20](https://peps.python.org/pep-0020/)

> There should be one-- and preferably only one --obvious way to do it.

Reply:
I agree. The only reason I added this was to go back to the original implementation if `transfer_rates` is set. In an ideal world we would just do the explicit factorization, but it doesn't seem to be working correctly when transfer rates are used.

Reply:
Part of me wonders if CRAM is not the correct method for solving the coupled block matrices in the transfer rates case, but if anybody has insight on that it would be great.

Reply:
hi @eepeterson, this seems to be a good improvement! 

Just curious to know what kind of errors did you notice when using LU with **transfer_rates**? I've also spent a bit of time recently trying to think of alternative methods we could used to solve the block matrix when many transfer rates are set for speeding it up, and LU factorization came also out. 

Note also that the block matrix is only constructed when a **destination_material**  is set in the **add_transfer_rate** method,  otherwise it's the same depletion matrix as before with only a removal term added. Did you get strange results with or without a **destination_material** set?

Reply:
Thanks for chiming in @church89. I'm seeing strange results primarily in the `assert_reaction_rates_equal` portion of the `deplete_with_transfer_rates` regression tests. The most finicky test is the final one in the suite `"depletion_with_transfer"` so the `destination_material` is set. However, sometimes other ones fail as well. A couple concerns I have are below:

1. Adding external feed and removal seems to introduce eigenvalues with positive real parts in the burnup matrix which is not where CRAM's approximation is accurate. They may be small enough with low feed rates and short time steps to not make a big difference, but it's still a concern. Until I can dig into the theory a bit more it seems like CRAM is ill-suited for this block matrix coupled approach for transfer rates. By contrast, the eigenvalues of every burnup matrix evaluated in the `deplete_with_transport` regression test suite all have negative real parts.
2. Despite the theoretical justification that CRAM shouldn't be used for these problems, the resulting nuclide densities coming out of CRAM after the first timestep agree between develop and this branch to within a relative tolerance of 1e-14. It's when they get put back in to the simulation with `openmc.lib` and resulting reaction rates are calculated that they disagree with the reference results.
3. The tolerances on the nuclide concentrations and reaction rates seem very small for simulations with 500 particles and 2 batches. By contrast, the tests in the `deplete_no_transport` case have atom tolerances that are on the order of 1e-3 and reaction rate tolerances on the order of 1e-2 or 1e-1 almost. Right now the tolerances in `deplete_with_transfer_rates` are 1e-4 and 1e-5 respectively. At the very least it makes sense to me that the reaction rate tolerances should be larger than the atom tolerances. Were there justifications for the tolerances in this test suite or could they be changed to be more in line with the other suites?

Reply:
@eepeterson, for the **transfer_rates** case without **destination_material** set, I would have excepted not to give any issues, independently on the transfer rates values set, as it should simply add a proportional term  to the diagonal terms of the burnup matrix and no block matrix is built. 

On the other hand, when the destination materials are set, the coupled block matrix becomes very sparse and might introduce some instability in the  CRAM, but apart from a bit more negative nuclide densities results made sense. 

There is no particular reason for the tolerances used, so if it helps pass some more tests and be in line with other suites we can change them.